alicloud/resource_alicloud_cen_child_instance_attachment_test.go

```diff
@@ -0,0 +1,185 @@
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCenChildInstanceAttachmentMulti,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCenChildInstanceAttachmentExists("alicloud_cen_child_instance_attachment.bar1", &instance),
+					resource.TestCheckResourceAttr("alicloud_cen_child_instance_attachment.bar1", "child_instance_type", "VPC"),
+					resource.TestCheckResourceAttr("alicloud_cen_child_instance_attachment.bar1", "child_instance_region_id", "cn-beijing"),
+					testAccCheckCenChildInstanceAttachmentExists("alicloud_cen_child_instance_attachment.bar2", &instance),
+					resource.TestCheckResourceAttr("alicloud_cen_child_instance_attachment.bar2", "child_instance_type", "VPC"),
+					resource.TestCheckResourceAttr("alicloud_cen_child_instance_attachment.bar2", "child_instance_region_id", "cn-shanghai"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCenChildInstanceAttachmentExists(n string, instance *cbn.ChildInstance) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Cen Child Instance ID is set")
+		}
+
+		client := testAccProvider.Meta().(*AliyunClient)
+
+		instanceId, cenId, err := getCenIdAndAnotherId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		childInstance, err := client.DescribeCenAttachedChildInstanceById(instanceId, cenId)
+		if err != nil {
+			return err
+		}
+
+		if childInstance.Status != "Attached" {
+			return fmt.Errorf("CEN id %s instance id %s status error", cenId, instanceId)
+		}
+
+		*instance = childInstance
+		return nil
+	}
+}
+
+func testAccCheckCenChildInstanceAttachmentDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*AliyunClient)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "alicloud_cen_child_instance_attachment" {
+			continue
+		}
+
+		instanceId, cenId, err := getCenIdAndAnotherId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		instance, err := client.DescribeCenAttachedChildInstanceById(instanceId, cenId)
+
+		if err != nil {
+			if NotFoundError(err) {
+				continue
+			}
+			return err
+		}
+
+		if instance.ChildInstanceId != "" {
+			return fmt.Errorf("CEN %s child instance %s still attach", instance.CenId, instance.ChildInstanceId)
+		}
+	}
+
+	return nil
+}
+
+const testAccCenChildInstanceAttachmentConfig = `
+resource "alicloud_cen_instance" "cen" {
+	name = "terraform-01"
+	description = "terraform01"
+}
+
+resource "alicloud_vpc" "vpc" {
+	name = "terraform-01"
+	cidr_block = "192.168.0.0/16"
+}
+
+resource "alicloud_cen_child_instance_attachment" "foo" {
+    cen_id = "${alicloud_cen_instance.cen.id}"
+    child_instance_id = "${alicloud_vpc.vpc.id}"
+    child_instance_type = "VPC"
+    child_instance_region_id = "cn-beijing"
+}
+`
+
+const testAccCenChildInstanceAttachmentMulti = `
```

Comment:
I think you can add another test case which attaching child instance in the same region.